components/validator/guardian.go

```diff
@@ -0,0 +1,167 @@
+package validator
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	_ "net/http/pprof"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/kroma-network/kroma/bindings/bindings"
+	"github.com/kroma-network/kroma/components/node/eth"
+	"github.com/kroma-network/kroma/utils"
+	"github.com/kroma-network/kroma/utils/service/txmgr"
+)
+
+// Guardian is responsible for validating outputs
+type Guardian struct {
+	log      log.Logger
+	cfg      Config
+	ctx      context.Context
+	cancel   context.CancelFunc
+	callOpts *bind.CallOpts
+	txOpts   *bind.TransactOpts
+	wg       sync.WaitGroup
+
+	securityCouncilContract *bindings.SecurityCouncil
+	securityCouncilSub      ethereum.Subscription
+
+	validationRequestedChan chan *bindings.SecurityCouncilValidationRequested
+
+	txCandidatesChan chan<- txmgr.TxCandidate
+}
+
+// NewGuardian creates a new Guardian
+func NewGuardian(parentCtx context.Context, cfg Config, l log.Logger, txCandidateChan chan<- txmgr.TxCandidate) (*Guardian, error) {
+	securityCouncilContract, err := bindings.NewSecurityCouncil(cfg.SecurityCouncilAddr, cfg.L1Client)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(parentCtx)
+
+	return &Guardian{
+		log:                     l,
+		cfg:                     cfg,
+		ctx:                     ctx,
+		cancel:                  cancel,
+		callOpts:                utils.NewCallOptsWithSender(ctx, cfg.TxManager.From()),
+		txOpts:                  utils.NewSimpleTxOpts(ctx, cfg.TxManager.From(), cfg.TxManager.Signer),
+		securityCouncilContract: securityCouncilContract,
+		validationRequestedChan: make(chan *bindings.SecurityCouncilValidationRequested),
+		txCandidatesChan:        txCandidateChan,
+	}, nil
+}
+
+func (g *Guardian) Stop() error {
+	g.log.Info("stop Guardian")
+
+	if g.securityCouncilSub != nil {
+		g.securityCouncilSub.Unsubscribe()
+	}
+	close(g.validationRequestedChan)
+
+	g.cancel()
+	g.wg.Wait()
+	return nil
+}
+
+func (g *Guardian) handleValidationRequested(logs chan *bindings.SecurityCouncilValidationRequested) {
+	defer func() {
+		g.wg.Done()
+	}()
+	for {
+		select {
+		case ev := <-logs:
+			g.wg.Add(1)
+			go g.processOutputValidation(ev)
+		case <-g.ctx.Done():
+			return
+		}
+	}
+}
+
+func (g *Guardian) processOutputValidation(event *bindings.SecurityCouncilValidationRequested) {
+	ticker := time.NewTicker(time.Minute)
+	defer func() {
+		ticker.Stop()
+		g.wg.Done()
+	}()
+
+	for {
+	Loop:
+		select {
+		case <-ticker.C:
+			isValidate, err := g.isL2OutputValidate(event.OutputRoot, event.L2BlockNumber.Uint64())
+			if err != nil {
+				g.log.Error("isL2OutputValidate failed", "err", err)
```

Comment:
Let's provide details
```suggestion
				g.log.Error("isL2OutputValidate failed", "err", err, "l2BlockNumber", event.L2BlockNumber.Uint64())
```